Feature/9763 update medication wording for va video

juancstlm-a6 opened the pull request:
## Description of Change
Update the content in appointment details for the listed types to include the following component when  `Upcoming` and `Cancelled` status. Existing content/wording relating to medication wording will be removed in favor of only the component listed below. 
- Video Atlas
- Video GFE
- Video Home

## Screenshots/Video

https://github.com/user-attachments/assets/cff54897-2091-4df6-959c-be554f9a4938 

https://github.com/user-attachments/assets/e981a9cc-ce92-4133-9024-f78bcf5be7b7 

| iOS | Android |
| --- | --- |
| <img width="100%" alt="Screenshot 2024-10-23 at 10 15 42 AM" src="https://github.com/user-attachments/assets/68603c09-5e1c-4a1e-90d1-140986cfd88e">|<img width="100%" alt="Screenshot 2024-10-17 at 5 16 33 PM" src="https://github.com/user-attachments/assets/4365868c-b18f-47cb-8a68-b54e7b969377">|



## Testing
<!-- What testing was done to verify the changes (local/unit)? What testing remains? Note edge cases, or special
situations that could not be tested during development. -->

- [X] Tested on iOS 
- [X] Tested on Android 

## Reviewer Validations
- [ ] Update Video Atlas Appointment details to include the "Medication wording" component for `Upcoming` and `Cancelled` status
- [ ] Update GFE Appointment details to include the "Medication wording" component `Upcoming` and `Cancelled` status
- [ ] Update Video Home Appointment details to include the "Medication wording" component `Upcoming` and `Cancelled` status
-  [ ] "Medication wording" component contains a link that opens https://www.va.gov/resources/what-should-i-bring-to-my-health-care-appointments/ in a in app web view
- [ ] "Medication wording" component for a Video appointment with type `At Home` `GFE` `Atlas` has a link to open a sheet that displays how to set up your device. 
  ![Image](https://github.com/user-attachments/assets/e2826886-769d-452d-b9f4-b009615a73ea)
- [ ] Update PrepareForVideoVisit component to include only the updated content in the screenshot above [Figma](https://www.figma.com/design/ZIHY63cFlD7pFjqQLA5rFN/Appointments-2.0---%F0%9F%94%8D-Working---VA-Mobile?node-id=3190-2705&t=5W7oLepJvCN8iVFf-4)
- [ ] e2e Test are updated or created to cover the changes
- [ ] Unit tests are updated or create to cover the changes

## PR Checklist
<!-- Engineer: make sure all these items are checked off before requesting a review -->
  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [X] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

juancstlm-a6 commented on VAMobile/e2e/tests/AppointmentsExpanded.e2e.ts:
```diff
@@ -532,10 +582,6 @@ export async function apppointmentVerification(pastAppointment = false) {
     if (!pastAppointment) {
       await expect(element(by.text('Video appointment')))
       await expect(element(by.text('You can join 30 minutes before your appointment time.'))).toExist()
-      await expect(element(by.id('prepareForVideoVisitTestID'))).toExist()
-      await element(by.id('prepareForVideoVisitTestID')).tap()
-      await expect(element(by.text('Appointments help'))).toExist()
-      await element(by.text('Close')).tap()
```
This check is now part of checkMedicationWording

juancstlm-a6 commented on VAMobile/src/screens/HealthScreen/Appointments/UpcomingAppointments/PrepareForVideoVisit/PrepareForVideoVisit.test.tsx:
```diff
@@ -13,44 +14,14 @@ context('PrepareForVideoVisit', () => {
   it('initializes correctly', () => {
-    expect(screen.getByText('Before your appointment:')).toBeTruthy()
-    expect(
-      screen.getByText(
-        'If you’re using an iPad or iPhone for your appointment, you’ll need to download the VA Video Connect iOS app beforehand. If you’re using any other device, you don’t need to download any software or app before your appointment.',
-      ),
-    ).toBeTruthy()
-    expect(
-      screen.getByText(
-        'You’ll need to have access to a web camera and microphone. You can use an external camera and microphone if your device doesn’t have one.',
-      ),
-    ).toBeTruthy()
-    expect(
-      screen.getByText(
-        /To connect to your Virtual Meeting Room at the appointment time, click the "Join session" button or the link that's in your confirmation email./,
-      ),
-    ).toBeTruthy()
-    expect(screen.getByText('To have the best possible video experience, we recommend you:')).toBeTruthy()
-    expect(
-      screen.getByText('Connect to your video appointment from a quiet, private, and well-lighted location.'),
-    ).toBeTruthy()
-    expect(
-      screen.getByText('Check to ensure you have a strong Internet connection before your appointment.'),
-    ).toBeTruthy()
-    expect(
-      screen.getByText(
-        'Connect to your appointment using a Wi-Fi network if using your mobile phone, rather than your cellular data network.',
-      ),
-    ).toBeTruthy()
-    expect(screen.getByText('Medication review:')).toBeTruthy()
-    expect(
-      screen.getByText(
-        'During your video appointment, your provider will want to review all the medications, vitamins, herbs, and supplements you’re taking—no matter if you got them from another provider, VA clinic, or local store.',
-      ),
-    ).toBeTruthy()
-    expect(
-      screen.getByText(
-        "Please be ready to talk about your medications during your video visit to ensure you're getting the best and safest care possible.",
-      ),
-    ).toBeTruthy()
```
Figured since most of this was changing I should update this test to use only the translation function

Sparowhawk commented on VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentMedicationWording.tsx:
```diff
@@ -20,42 +31,72 @@ type AppointmentMedicationWordingProps = {
-          )
-        default:
-          return <></>
+      const conent = getContent()
```
there is a typo here

juancstlm-a6 replied:
Doh! Updated

Sparowhawk reviewed (changes requested):
just the typo, but everything else seems good

mistymg commented:
Posting what I had in Slack here, too:

1. I noticed the iOS screenshot includes some additional wording in the 3rd bullet ("or the link that's in your confirmation email") that the Android screenshot doesn't have. Which is correct? (Videos seemed ok though.)
2. Wondering about your choice to include a loading component for when user goes to web view.
-If a loading component is necessary, I'd suggest more explicit wording like Loading [VA.gov](http://va.gov/) webpage.... The 3 periods are necessary.

Sparowhawk reviewed (approved):
all good from what I can tell, just Misty's question left about loading message for the webview is all I think needs to be addressed

juancstlm-a6 commented:
Updated outdated screenshot for iOS
- Updated loading copy to "Loading VA.gov webpage..."
![Screenshot 2024-10-23 at 10 23 21 AM](https://github.com/user-attachments/assets/f1eea8a1-11d7-434f-9171-510dc25933a6)

https://github.com/user-attachments/assets/066d1d92-7fbc-4c8c-b02f-106698de8ccd
